include/CLI/App.hpp

```diff
@@ -490,157 +503,191 @@ class App {
+                        if(!var.empty()) {
+                            values.emplace_back();
+                            retval &= detail::lexical_cast(var, values.back());
+                        }
+                    }
+                } else {
+                    values.emplace_back();
+                    retval &= detail::lexical_cast(elem, values.back());
+                }
             }
             if(retval) {
                 return func(values);
             }
             return retval;
         };
 
-        Option *opt = add_option(option_name, std::move(fun), description, false);
+        Option *opt = add_option(option_name, std::move(fun), std::move(option_description), false);
         opt->type_name(detail::type_name<T>())->type_size(-1);
         return opt;
     }
 
     /// Set a help flag, replace the existing one if present
-    Option *set_help_flag(std::string flag_name = "", std::string description = "") {
+    Option *set_help_flag(std::string flag_name = "", const std::string &flag_description = "") {
+        // take flag_description by const reference otherwise add_flag tries to assign to flag_description
         if(help_ptr_ != nullptr) {
             remove_option(help_ptr_);
             help_ptr_ = nullptr;
         }
 
         // Empty name will simply remove the help flag
         if(!flag_name.empty()) {
-            help_ptr_ = add_flag(flag_name, description);
+            help_ptr_ = add_flag(flag_name, flag_description);
             help_ptr_->configurable(false);
         }
 
         return help_ptr_;
     }
 
     /// Set a help all flag, replaced the existing one if present
-    Option *set_help_all_flag(std::string help_name = "", std::string description = "") {
+    Option *set_help_all_flag(std::string help_name = "", const std::string &help_description = "") {
+        // take flag_description by const reference otherwise add_flag tries to assign to flag_description
         if(help_all_ptr_ != nullptr) {
             remove_option(help_all_ptr_);
             help_all_ptr_ = nullptr;
         }
 
         // Empty name will simply remove the help all flag
         if(!help_name.empty()) {
-            help_all_ptr_ = add_flag(help_name, description);
+            help_all_ptr_ = add_flag(help_name, help_description);
             help_all_ptr_->configurable(false);
         }
 
         return help_all_ptr_;
     }
 
-    /// Add option for flag
-    Option *add_flag(std::string flag_name, std::string description = "") {
-        CLI::callback_t fun = [](CLI::results_t) { return true; };
-        Option *opt = add_option(flag_name, fun, description, false);
-        if(opt->get_positional())
-            throw IncorrectConstruction::PositionalFlag(flag_name);
+  private:
+    /// internal function for adding a flag
+    Option *add_flag_internal(std::string &flag_name, CLI::callback_t &fun, std::string &flag_description) {
```

Comment:
This should start with an underscore, internal functions all start with underscore. (Python style)